### DISCUSSION TITLE
UGS won't connect to CNC - jserialcomm error

### DISCUSSION THREAD

**Discussion by reillyblum** [winder/Universal-G-Code-Sender#2316] [Help]
When I try to connect UGS to my Genmitsu 4040 pro, I get this jserial comm error message "error opening connection. could not connect to controller on port"

I'm running UGS on Mac OS Ventura 13.2
I'm not sure what's causing it — I've been checking and refreshing ports, quitting and reloading the app, jumping through all of the hoops, etc. Every now and then (about 5% of the time), the machine will connect just fine and I'm able run cuts without any problems.
<img width="466" alt="Screenshot 2023-09-11 at 6 07 17 PM" src="https://github.com/winder/Universal-G-Code-Sender/assets/63164417/a9b33feb-302f-4b2f-91c2-f4a7436cdbc6">

This is my first time running my own machine, so would really appreciate any help. thanks

**Comment by breiler** [comment]
Which version of UGS are you running?

**Reply by reillyblum** [reply]
2.0.14

**Reply by breiler** [reply]
Ok, that version still uses the old initiation protocol and should not be a problem.

You could give the latest 2.0.20 version a try:
https://github.com/winder/Universal-G-Code-Sender#downloads

**Reply by timbustad** [reply]
Hello... I'm getting something similar

Running newest version of UGS.  2.1.4   When trying to connect I get this
*** Connecting to jserialcomm://COM4:115200
*** Fetching device status
>>> ?
*** Fetching device status (2 of 3)...
>>> ?
*** Fetching device status (3 of 3)...
>>> ?
**** Connection closed ****

I have rebooted everything many times and still get the same message.  Checked all the cables,  Tired a older version as well.  

HELP Please!!

Thanks
tim

**Reply by breiler** [reply]
@timbustad please create a new thread with information about your controller. What version of the firmware you are using and how you installed it.

**Reply by timbustad** [reply]
Thanks @Breiler.  Will do

**Reply by danielHPCX** [reply]
I was facing same problem with UGS v2.1.7. I was using the CNC3018 (GRBL v0.9) with Offline controller connected. For some reason the controller was causing this issue. When i disconected the controller it worked.
![WhatsApp Image 2024-06-22 at 10 27 34](https://github.com/winder/Universal-G-Code-Sender/assets/53661528/696a45ca-50e1-49b8-8c39-338e704c217b)

**Reply by breiler** [reply]
The problem with the offline controller is documented here:
https://github.com/winder/Universal-G-Code-Sender/wiki/Troubleshooting-Overview#an-unexpected-command-was-completed-by-the-controller

**Comment by dastultz** [comment]
Just getting started with UGS after some years off. Turns out the latest
posting is the same problem I'm having.

I see timbustad was asked to start a new thread, I can do so as well, but I
first wanted to add something that might be helpful.

I get the same "Fetching device status" 3-count sequence and it fails to
connect. I then go to a low-level terminal program (I'm on Linux, I used
"screen") and connect successfully. I disconnect, then UGS connects without
issue. (I'm running GRBL on an Arduino which hasn't changed in forever but
I am on a new computer now.)

**Reply by breiler** [reply]
Welcome back

Yes please start a new thread, because the original post was about a specific error message that can happen if the port was opened with another program or if the user don't have the proper user rights to the device. I would prefer having your problem in it's own thread to make things less muddy.

I would like your GRBL version and settings. I'd appreciate if you can turn on verbose logging in the console and then connect to get all the responses from the controller. It may also help to get the application log (go to the menu Tools / Open log directory).

**Comment by giacummo** [comment]
hello

the same issue, in a laptop with linux just installed and just usg running... the las version,

**Reply by breiler** [reply]
Please create a new thread with more details.

**Comment by giacummo** [comment]
It is a problem with file permission...did you solve it?

**Comment by Hacarun** [comment]
Can any give the solution for my problem ?

This error occurred in ugs to microcontroller connection error 

Error opening connection: Could not connect to controller on port jserialcomm://ttyACM0:115200

**Reply by breiler** [reply]
Please create a new issue with more details.

I need to know what operating system you are using.
What controller you are connecting to and what firmware (and version) you have installed. 
I also need to know which version of UGS you are using.

I have a hunch that you are using Linux and don't have permission accessing the serial device:
https://support.arduino.cc/hc/en-us/articles/360016495679-Fix-port-access-on-Linux

**Comment by breiler** [comment]
I'm locking this discussion to avoid getting "I also have this problem" follow-up questions which are not related.

If you encounter a similar problem please create a new discussion with details about your system.
